### PR TITLE
20 node quadratic hexahedral element

### PR DESCRIPTION
**Pull request by kinnala**
This begins the work on #327 .

Edit: Includes also a feature for merging two meshes via `__add__`.

### CODE REVIEW

**Comment by kinnala** [comment]
The convergence rate seems to be lower than I expect; I'm not sure what causes this. Need to investigate a bit further.

**Comment by kinnala** [comment]
Added several new tests but couldn't find the culprit: still the new element converges as O(h^1/2) in H^1 norm and O(h) in L^2 norm which isn't O(h^2) and O(h^3) as I'd expect.

**Comment by kinnala** [comment]
Figured out that there could still be a hidden bug in the facet->edge expansion. Added a test in the latest commit.

**Comment by kinnala** [comment]
Yes! Figured it out. The facet->edge expansion that was fixed in #83 was working only in case of tetrahedral elements.

**Comment by kinnala** [comment]
Now we have the first quadratic hexahedral element. Do we still need the 27 node element? Maybe for Stokes?

File: skfem/element/element_hex/element_hex_s2.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import numpy as np
+
+from ..element_h1 import ElementH1
+from ...mesh.mesh3d import MeshHex
+
+
+class ElementHexS2(ElementH1):
```
Comment (gdmcbain) [inline skfem/element/element_hex/element_hex_s2.py:7]: `ElementQuadS2` is derived from `ElementQuad2` in which case for consistency `ElementHexS2` should be derived from `ElementHex2`, but perhaps it shouldn't have?   I've sometimes read that object-oriented inheritance is supposed to mean 'is a' but don't know how valid or important that is.  Anyway I think the tree should be consistent between quadrilaterals and hexahedra here.

File: skfem/element/element_hex/element_hex_s2.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    nodal_dofs = 1
+    edge_dofs = 1
+    dim = 3
+    maxdeg = 3
```
Comment (gdmcbain) [inline skfem/element/element_hex/element_hex_s2.py:11]: Is maxdeg = 3?  I would have thought it was 4, like `ElementHex2`; the eight `nodal_dofs` are quartic.

**Comment by gdmcbain** [comment]
> Do we still need the 27 node element? Maybe for Stokes?

Well we probably want it for completeness anyway and yes for Stokes, I think.  `ElementHex2` is often recommended for Stokes (e.g. Ern & Guermond 2004, _Theory and Practice of Finite Elements_, figure. 4.4), but I don't think I've seen serendipity elements used for incompressible fluids.

File: skfem/element/element_hex/element_hex_s2.py (hunk 2 above)
Comment (kinnala) [reply skfem/element/element_hex/element_hex_s2.py]: Increased this to 4. Also decreased to 3 in `ElementQuadS2`.

File: skfem/element/element_hex/element_hex_s2.py (hunk 1 above)
Comment (kinnala) [reply skfem/element/element_hex/element_hex_s2.py:7]: I changed how `ElementQuadS2` gets defined.

**Comment by kinnala** [comment]
> Well we probably want it for completeness anyway and yes for Stokes, I think.

Sure. That can be my next follow up task.

**Pull request merged**